clean_old_aggregates.py

```diff
@@ -0,0 +1,70 @@
+import keystoneauth1
+import datetime
+from keystoneauth1.identity import v3
+from keystoneauth1 import loading
+from keystoneauth1 import session
+from novaclient import client as nova_client
+from blazarclient import client as blazar_client
+
+def main():
+
+    loader = loading.get_plugin_loader('password')
+    auth = loader.load_from_options(auth_url='http://REDACTED:35357',
+    username='admin',
+    password='REDACTED',
+    project_id='REDACTED',
+    user_domain_id ='default')
+
+    sess = session.Session(auth=auth)
+
+    nova = nova_client.Client(version='2', region_name='CHI@TACC', session=sess)
+    blazar = blazar_client.Client(1, session=sess, service_type='reservation', region_name='CHI@TACC')
+
+    term_leases=[]
+    agglist=[]
+    aggdetails=[]
+    aggnames=[]
+    exp_aggs=[]
+    agg_host=[]
+
+    # Find leases with end date before current time
+    leases=blazar.lease.list()
+
+    time=datetime.datetime.now()
+    for x in leases:
+        lease_time=datetime.datetime.strptime(x['end_date'], '%Y-%m-%dT%H:%M:%S.%f')
+        if lease_time < time:
+            print("lease ending {} has ended. Time {}".format(lease_time,time))
+            term_leases.append(x['reservations'][0]['id'])
```

Comment:
This whole `for` statement can be elegantly expressed in Python using a [list comprehension](https://docs.python.org/3/tutorial/datastructures.html#list-comprehensions). It would look sort of like this:

```python
now = datetime.datetime.nowutc()

def is_terminated(lease):
  dt_fmt = '%Y-%m-%dT%H:%M:%S.%f'
  return now > datetime.datetime.strptime(lease['end_date'], dt_fmt).replace(tzinfo=timezone('UTC'))

term_leases = [lease for lease in leases if is_terminated(lease)]
```

`term_leases` then contains all the lease objects where `end_date` is in the past. This is utilizing an inline helper function, which can be helpful in cases like this.

In general I recommend keeping the thing you're operating over (leases) as intact as possible. Rather than finding the reservation ID and keeping only that, we maintain a reference to the lease object. This is because the operation "find the aggregate for this lease" is best expressed as a function of one parameter: the lease itself. This allows a lot of flexibility as to how the lease is related to the Nova aggregate under the hood. Right now the name of the aggregate is linked with the reservation ID under the lease, but it could feasibly change in any number of ways. But, if it changes, only one function should need to be modified here.